### DISCUSSION TITLE
CS0310	'TModel' must be a non-abstract type

### DISCUSSION THREAD

**Discussion by DWelter24** [dotnetcore/BootstrapBlazor#4101] [Q&A] [answered]
i am unable to compile the latest code from github, or by downloading 8.8.3 zip.  this error is reported in 5 files.

CS0310	'TModel' must be a non-abstract type with a public parameterless constructor in order to use it as parameter 'TModel' in the generic type or method 'DataServiceBase<TModel>'	BootstrapBlazor.DataAccess.EntityFrameworkCore (net6.0), BootstrapBlazor.DataAccess.EntityFrameworkCore (net7.0), BootstrapBlazor.DataAccess.EntityFrameworkCore (net8.0)	C:\Users\......\GitHub\BootstrapBlazor\src\Extensions\DataServices\BootstrapBlazor.DataAccess.EntityFrameworkCore\DefaultDataService.cs

**Comment by ArgoZhang** [comment]
@DWelter24 let me check it. later reply you. sorry for that

**Reply by ArgoZhang** [reply]
@DWelter24 Hi. that is my fault. please use git clone the `fix-build` branch. github discussion can't upload a package more than 25M.

i will merge this branch into main if there is no error.

**Reply by ArgoZhang** [reply, marked as answer]
@DWelter24 link issue 
#4014 
#4015

**Reply by ArgoZhang** [reply]
@DWelter24 now you can `git clone` the main branch code
